### ISSUE TITLE
Avoid unnecessary logging on anonymous access.

### ISSUE THREAD

**Issue by ksauzz** [basho/riak_cs#876]
We shouldn't output no_user_key error on proper anonymous access such as getting a public object.

```
2014-05-29 18:08:35.835 [error] <0.2515.0>@riak_cs_wm_common:maybe_create_user:218 Retrieval of user record for s3 failed. Reason: no_user_key
```

**Comment by shino** [comment]
For example, if one use riak_cs for file download with public read ACL, this log can happen heavily. Lager, by default setting, calls `file:datasync/1` for error or higher severity [1]. So this issue may increase disk load.

[1] https://github.com/basho/lager/blob/master/src/lager_file_backend.erl#L228

**Comment by shino** [comment]
Move to 1.5.0

**Comment by shino** [comment]
Resolved by #883.
